openlibrary/core/bookshelves.py

```diff
@@ -21,22 +23,24 @@ class Bookshelves(db.CommonExtras):
     }
 
     @classmethod
-    def summary(cls):
+    def summary(cls) -> dict[str, dict[str, int]]:
         return {
             'total_books_logged': {
-                'total': Bookshelves.total_books_logged(),
-                'month': Bookshelves.total_books_logged(since=DATE_ONE_MONTH_AGO),
-                'week': Bookshelves.total_books_logged(since=DATE_ONE_WEEK_AGO),
+                'total': Bookshelves.total_books_logged() or 0,
+                'month': Bookshelves.total_books_logged(since=DATE_ONE_MONTH_AGO) or 0,
+                'week': Bookshelves.total_books_logged(since=DATE_ONE_WEEK_AGO) or 0,
             },
             'total_users_logged': {
-                'total': Bookshelves.total_unique_users(),
-                'month': Bookshelves.total_unique_users(since=DATE_ONE_MONTH_AGO),
-                'week': Bookshelves.total_unique_users(since=DATE_ONE_WEEK_AGO),
+                'total': Bookshelves.total_unique_users() or 0,
+                'month': Bookshelves.total_unique_users(since=DATE_ONE_MONTH_AGO) or 0,
+                'week': Bookshelves.total_unique_users(since=DATE_ONE_WEEK_AGO) or 0,
             },
         }
 
     @classmethod
-    def total_books_logged(cls, shelf_ids=None, since=None):
+    def total_books_logged(
+        cls, shelf_ids: Optional[list[str]] = None, since: Optional[date] = None
```

Comment:
`mypy` automatically adds `Optional` if the default value is set to None :+1: Ditto elsewhere

```suggestion
        cls, shelf_ids: list[str] = None, since: date = None
```